src/analysis/asymptotics/superpolynomial_decay.lean

```diff
@@ -0,0 +1,252 @@
+/-
+Copyright (c) 2021 Devon Tuma. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Devon Tuma
+-/
+import analysis.asymptotics.asymptotics
+import analysis.asymptotics.specific_asymptotics
+import data.polynomial.eval
+
+/-!
+# Super-Polynomial Function Decay
+
+This file defines a predicate `asymptotics.superpolynomial_decay f` for a function satisfying
+  one of following equivalent definitions (The definition is in terms of the first condition):
+
+* `f` is `O(x ^ c)` for all (or equivalently sufficiently small) integers `c`
+* `f` is `O(p(x)⁻¹)` for all (or equivalently sufficiently large) polynomials `p`
+* `p(x) * f` is bounded for all polynomials `p`
+* `p(x) * f` tends to `𝓝 0` for all polynomials `p`
+
+The definition of superpolynomial decay for a function `f : α → 𝕜`
+  is made relative to an algebra structure `[algebra α 𝕜]`.
+Super-polynomial decay then means the function `f x` decays faster than
+  `(p.eval (algebra_map α 𝕜 x))⁻¹` for all polynomials `p : polynomial 𝕜`.
+
+When the algebra structure is given by `n ↦ ↑n : ℕ → ℝ` this defines negligible functions:
+https://en.wikipedia.org/wiki/Negligible_function
+
+When the algebra structure is given by `(r₁,...,rₙ) ↦ r₁*...*rₙ : ℝⁿ → ℝ` this is equivalent
+  to the definition of rapidly decreasing functions given here:
+https://ncatlab.org/nlab/show/rapidly+decreasing+function
+-/
```

Comment:
Can you add a `## Main results` section, and highlight a few of the keys lemmas?